packages/site/src/components/SkipToArticle.tsx

```diff
@@ -0,0 +1,52 @@
+import { useCallback } from 'react';
+
+function makeSkipClickHander(hash: string) {
+  return (e: React.UIEvent<HTMLElement, Event>) => {
+    e.preventDefault();
+    const el = document.querySelector(`#${hash}`);
+    if (!el) return;
+    el.scrollIntoView({ behavior: 'smooth' });
+    history.replaceState(undefined, '', `#${hash}`);
+    (el.nextSibling as HTMLElement).focus({ preventScroll: true });
+    (e.target as HTMLElement).blur();
+  };
+}
+
+export function SkipToArticle({
+  frontmatter = true,
+  article = true,
+}: {
+  frontmatter?: boolean;
+  article?: boolean;
+}) {
+  const fm = 'skip-to-frontmatter';
+  const art = 'skip-to-article';
+
+  const frontmatterHander = useCallback(() => makeSkipClickHander(fm), [frontmatter]);
+  const articleHandler = useCallback(() => makeSkipClickHander(art), [article]);
+  return (
+    <div
+      className="fixed top-1 left-1 h-[0px] w-[0px] focus-within:z-40 focus-within:h-auto focus-within:w-auto bg-white overflow-hidden focus-within:p-2 focus-within:ring-1"
```

Comment:
need to look at using `focus-visible` here so that these focus states only apply to keyboard navigation

Reply:
You can use the `sr-only` class here! That picks up a bunch of these I think (the top/h/w at least)?

Reply:
Ok, I think in this case yours is better because we are changing things!

Reply:
skip nav is for keyboard users rather than screen readers only